lib/Doctrine/DBAL/Schema/OracleSchemaManager.php

```diff
@@ -146,32 +146,30 @@ protected function _getPortableTableColumnDefinition($tableColumn)
             $tableColumn['data_default'] = trim($tableColumn['data_default'], "'");
         }
 
-        $precision = null;
-        $scale     = null;
+        if ($tableColumn['data_precision'] !== null) {
+            $precision = (int) $tableColumn['data_precision'];
+        }
+
+        if ($tableColumn['data_scale'] !== null) {
+            $scale = (int) $tableColumn['data_scale'];
+        }
 
         $type                    = $this->_platform->getDoctrineTypeMapping($dbType);
         $type                    = $this->extractDoctrineTypeFromComment($tableColumn['comments'], $type);
         $tableColumn['comments'] = $this->removeDoctrineTypeFromComment($tableColumn['comments'], $type);
 
         switch ($dbType) {
             case 'number':
-                if ($tableColumn['data_precision'] === 20 && $tableColumn['data_scale'] === 0) {
-                    $precision = 20;
-                    $scale     = 0;
-                    $type      = 'bigint';
-                } elseif ($tableColumn['data_precision'] === 5 && $tableColumn['data_scale'] === 0) {
-                    $type      = 'smallint';
-                    $precision = 5;
-                    $scale     = 0;
-                } elseif ($tableColumn['data_precision'] === 1 && $tableColumn['data_scale'] === 0) {
-                    $precision = 1;
-                    $scale     = 0;
-                    $type      = 'boolean';
+                if ($precision === 20 && $scale === 0) {
+                    $type = 'bigint';
+                } elseif ($precision === 5 && $scale === 0) {
+                    $type = 'smallint';
+                } elseif ($precision === 1 && $scale === 0) {
+                    $type = 'boolean';
                 } elseif ($tableColumn['data_scale'] > 0) {
```

Comment:
Nice clean up :+1: 

This could be `$scale > 0` (not that it would make a big difference :stuck_out_tongue:)